obsolete/weather/weather/wm918/wx200d-1.2/wxstdout.1
.\" @(#)$Id: wxstdout.1,v 1.1 2002/09/23 19:12:51 bogdan Rel $
.\"
.\" Copyright (C) 1998 - 2000 Tim Witham <twitham@quiknet.com>
.\"
.\" (see the files README and COPYING for more details)

.TH WXSTDOUT 1 "Feb 01, 2000" "wx200d Manual"
.SH NAME
wxstdout \- send a copy of binary wx200d data to stdout
.SH SYNOPSIS
.B wxstdout
[-h] [[host][:port]]

.SH DESCRIPTION

.B wxstdout
is an extremely simple
.B wx200d(8)
weather client that sends a copy of the binary WX200 data to the
standard output.  This is not useful alone but is when combined with a
WX200 data filter program such as
.B wxfilter(1)
or
.B wxdebug(1).
.P

.B wxstdout
connects to the given host and port.  If host or port is not
specified, the
.B WX200D
environment variable is checked for the missing field(s).  Finally, if
neither of these provide the host or port, the built-in defaults
(localhost:9753) are used.
.P

.SH OPTIONS

.TP 0.5i
.B -h
Display a help usage message showing these options, then exit.

.SH ENVIRONMENT

.TP 0.5i
.B WX200D
If set, the [host][:port] to contact if none is specified on the
command line.

.SH EXAMPLES

.TP 0.5i
.B wxstdout | wxfilter
Monitor the current WX200 data in tab-delimited format.

.TP 0.5i
.B wxstdout -m foo | wxfilter
Monitor the current tab-delimited data of the WX200 on machine foo.

.TP 0.5i
.B wxstdout | wxdebug
Monitor the content of all WX200 data groups.

.SH LIMITATIONS

.SH BUGS
Not likely.

.SH SEE ALSO
wxfilter(1), wxdebug(1), wx200d(8), wx200d(5)

.SH AUTHOR
Tim Witham <twitham@quiknet.com>
.SH COPYRIGHT
Copyright (C) 1998 - 2000 Tim Witham <twitham@quiknet.com>

wxstdout is released under the conditions of the GNU General Public
License.  See the files README and COPYING in the distribution for
details.


sip_router/NEWS
XXX content of this file reflects a future release;
    features described here may or may not be available
    in the CVS tree
    

Release notes for SIP Express Router (ser)
***********************************************

$Id: NEWS,v 1.12 2003/04/18 14:11:30 jiri Exp $

***********************************************
* Changes introduced in 0.8.11
***********************************************

+--------------------------------------------------------+
| CAUTION: the 0.8.11 release include changes which      |
| are incompatible with scripts and databases used       |
| in previous versions. Care is advised when upgrading   |
| from previous releases to 0.8.11.                      |
+--------------------------------------------------------+

New features
=============
- RFC3261 support
	- TCP support and cross-transport forwarding [core]
	- loose routing support [rr module]
- New modules
	- vm -- voicemail interface [vm]
	- ENUM support [enum]
	- presence agent [pa]
	- dynamic domain management -- allows to manipulate 
  	  hosting of multiple domains in run-time [module]
	- flat-text-file database support [dbtext]
	- rich access control lists [permissions]
- Feature Improvements
	- click-to-dial, which is based on improved tm/FIFO 
	  that better supports external applications [tm module]
	- web accounting -- acc module can report to serweb
  	  on placed calls [acc module]
	- improved exec module (header fields passed now
      as environment variables to scripts) [exec module]
- Architectural Improvements
	- powerpc fast locking support
	- netbsd support
	- 64 bits arch. support (e.g. netbsd/sparc64).
- New Experimental Features (not tested at all yet)
	- nathelper utility for Cisco/ATA NAT traversal [nathelper]
	- another NAT traversal utility [mangler]
	- postgress support [postgress]
	- fcp module [fcp]
	- pdt module (prefix2domain) [pdt]

Changes to use of ser scripts
=============================

About Multiple Transport Support
--------------------------------
SER now suports multiple transport protocols: UDP and TCP. As there
may be UAs which support only either protocol and cannot speak to
each other directly, we recommend to alway record-route SIP requests,
to keep the transport-translating SER in path. Also, if a destination
transport is not known, stateful forwarding is recommended -- use of
stateless forwarding for TCP2UDP would result in loss of reliability.


core
----
- reply_route has been renamed to failure_route -- the old name caused
  too much confusion
- forward_tcp and forward_udp can force SER to forward via specific
  transport protocol

acc module:
-----------
- radius and sql support integrated in this module; you need to
  recompile to enable it
- acc_flag is now called log_flag to better reflect it relates
  to the syslog mode (as opposed to sql/radius); for the same
  reasons, the accounting action is now called "acc_log_request"
- log_fmt allows now to specify what will be printed to syslog

auth module:
------------
- auth module has been split in auth, auth_db, auth_radius, group
  group_radius, uri and uri_radius 
- all the parameters that were part of former auth module are now 
  part of auth_db module
XXX 

im module:
----------
- im is no longer used and has been obsoleted by TM

exec module:
------------
- exec_uri and exec_user have been obsoleted by exec_dset; 
  exec_dset is identical to exec_uri in capabilities; it 
  additionaly passes content of request elements (header 
  fields and URI parts) in environment variables; users of 
  exec_user can use exec_dset now and use the "URI_USER"  
  variable to learn user part of URI
- exec_dset and exec_msg return false, if return value of 
  script does not euqal zero
- exec_dset takes an additional parameter, which enables 
  validation of SIP URIs returned by external application

jabber module:
--------------
- presence support for Jabber users is enabled loading the PA
  module and using subscribe("jabber") for SUBSCRIBE requests to
  jabber user 

msilo module:
-------------
- m_store has now a parameter to set what should be considered
  for storing as destination uri. This enables support for saving
  the messages on negative replies.

radius_acc module:
------------------
- radius_acc module has been removed and radius accounting 
  is now part of acc module

registrar/usrloc modules:
-------------------------
XXX
  
rr module:
----------
- addRecordRoute has been replaced with record_route
- rewriteFromRoute has been replaced with loose_route()
- a new option, "enable_full_lr" can be set to make life
  with misimplemented UAs easier and put LR in from "lr=on"

tm module:
----------
- t_reply_unsafe, used in former versions within reply_routes,
  is deprecated; now t_reply is used from any places in script
- t_on_negative is renamed to t_on_failure -- the old name just
  caused too much confusion
- FIFO t_uac used by some applications (like serweb) has been
  replaced with t_uac_dlg (which allows easier use by dialog-
  oriented applications, like click-to-dial) 
- if you wish to do forward to another destination from 
  failure_route (reply_route formerly), you need to call t_relay
  or t_relay_to explicitely now
- t_relay_to has been replaced with t_relay_to_udp and t_relay_to_tcp



obsolete/weather/weather/wm918/wx200d-1.2/wxstdout.c
/*
 * @(#)$Id: wxstdout.c,v 1.1 2002/09/23 19:12:51 bogdan Rel $
 *
 * Copyright (C) 1998 - 2000 Tim Witham <twitham@quiknet.com>
 *
 * (see the files README and COPYING for more details)
 *
 * This file implements a client which sends a data stream to stdout.
 *
 */

#include <stdio.h>
#include <unistd.h>
#include <getopt.h>
#include <stdlib.h>
#include <string.h>
#include "wx200.h"

char *progname;

void
usage()
{
  printf("usage: %s [-h] [[host][:port]]
	-h	show this help and exit
", progname);
  exit(0);
}

int
main(int argc, char **argv)
{
  int c, socket;
  char *hostport = NULL;

  progname = strrchr(argv[0], '/');
  if (progname == NULL)
    progname = argv[0];
  else
    progname++;
  while ((c = getopt(argc, argv, "Hh")) != EOF) {
    switch(c) {
    case '?':
    case 'h':
    case 'H':			/* help */
      usage();
      break;
    }
  }
  if (optind < argc && argv[optind] != NULL)
    hostport = argv[optind];
  if ((socket = wx200open(hostport)) == -1)
    exit(1);
  while (read(socket, &c, 1) == 1) {
    write(1, &c, 1);
  }
  exit(wx200close(socket));
}
